sdk/browser/src/AffinityWallet.ts

```diff
@@ -18,6 +18,7 @@ export class AffinityWallet extends CoreNetwork {
 
     this.keysService = new KeysService(encryptedSeed, password)
     this.walletStorageService = new WalletStorageService(encryptedSeed, password, sdkOptions)
+    this._component = EventComponent.AffinidiBrowserSDK
```

Comment:
Do you mean sth like:

```ts
export default class MetricsService {
  _apiKey: string
  _metricsUrl: string
  _component: EventComponent

  constructor(options: MetricsServiceOptions, component: EventComponent) {
    this._apiKey = options.apiKey
    this._metricsUrl = options.metricsUrl
    this._component = component
  }

  send(event: MetricsEvent): void {
    const metricsEvent = Object.assign({}, event, { component: this._component })

    metrics.send(metricsEvent, this._apiKey, this._metricsUrl)
  }
}
```

If that's the case, I have some further questions:

Since `MetricsService` is only newed once in the core SDK, at the time it is constructed we won't know which SDK (browser/expo/rn) will actually use it so we won't be able to pass the component value yet.
Also `MetricsService` is not exported but the `CommonNetworkMember` is.
Do you suggest to export `MetricsService` (currently indeed a **readonly private** property of `CommonNetworkMember`) for other SDK (browser/expo/rn) to take over the instantiation?

Reply:
@DenisPopov15  nvm. I think I probably know how to do it.